app/web/src/workers/webworker.ts

```diff
@@ -936,34 +936,85 @@ const get = async (
       components,
     };
     return list;
-  } else if (kind === "ComponentConnectionsListBeta") {
-    const rawList = atomDoc as RawComponentConnectionsListBeta;
-    const maybeComponentConnections = await Promise.all(
+  } else if (kind === "IncomingConnections") {
+    // FIXME(nick): talked with John O and this is not working fully. We are close though. Do not
+    // be alarmed if you come here and go "oh shit it's not quite right" because it isn't.
+    const raw = atomDoc as RawIncomingConnections;
+    clear_weak_references(changeSetId, { kind, args: raw.id });
+
+    // First, get the current component.
+    const component = (await get(
+      workspaceId,
+      changeSetId,
+      "Component",
+      raw.id,
+    )) as BifrostComponent;
+    weak_reference(
+      changeSetId,
+      { kind: "Component", args: raw.id },
+      { kind, args: raw.id },
+    );
+
+    // Collect all weak references within the incoming connections.
+    const connections = await Promise.all(
+      raw.connections.map(async (c) => {
+        const fromComponent = (await get(
+          workspaceId,
+          changeSetId,
+          c.fromComponentId.kind,
+          c.fromComponentId.id,
+        )) as BifrostComponent;
+        const toComponent = (await get(
+          workspaceId,
+          changeSetId,
+          c.toComponentId.kind,
+          c.toComponentId.id,
+        )) as BifrostComponent;
+
```

Comment:
@jobelenus I forgot to remove this since we collect it once at the start. We'll need to look at weak references after this PR anyway but just wanted to call it out. It should just re-use the component from a couple of lines up (though we still need to mark the weak reference)